src/editor/codeeditor.cpp

```diff
@@ -181,10 +176,67 @@ void CodeEditor::lineNumberAreaPaintEvent(QPaintEvent *event)
         bottom += bottom - temp;// bottom - temp = dy which is block height
         ++blockNumber;
     }
+    mLinesCount = blockNumber;
 }
 
 void CodeEditor::saveStateInTheHistory()
 {
     std::string newFileState = this->toPlainText().toUtf8().constData();
-    changeManager->writeChange(newFileState);
+    mChangeManager->writeChange(newFileState);
+}
+
+void CodeEditor::mouseMoveEvent(QMouseEvent *event)
+{
+    if (event->button() == Qt::NoButton)
+    {
+        QTextBlock block = this->firstVisibleBlock();
+
+        int top = static_cast<int>(blockBoundingGeometry(block).translated(contentOffset()).top());//top of currentblock 0
+        int bottom = top + static_cast<int>(blockBoundingRect(block).height());
+        int side = bottom - top;// size of each side of comment button
+
+       // side of rectancge where our bottom will be. X-0 && Y-0 start from the left top, so (top < bottom)
+        const QSize buttonSize = QSize(side, side);
+        mAddCommentButton->setFixedSize(buttonSize);
+
+        auto currSliderPos = this->verticalScrollBar()->sliderPosition();
+
+        int commentAreaRightMargin = this->width() - this->verticalScrollBar()->width() - getLineNumberAreaWidth();
+        int commentAreaLeftMargin = commentAreaRightMargin - side;
+
+       if((event->x() >= commentAreaLeftMargin) && (event->x() <= commentAreaRightMargin))//mouse inside comment block
+       {
+            int linesFromTheTop = event->y() / side;
+            int currLine = linesFromTheTop + currSliderPos + 1;//because first block = 0
+            mAddCommentButton->setCurrentLine(currLine);
+            int commentBottonYpos = linesFromTheTop * side;// get Y pos for bottom
+
+            if(currLine <= mLinesCount)// check if the line exists
+            {
+                int commentBottonXpos = commentAreaLeftMargin + getLineNumberAreaWidth();
+
+                if(currSliderPos)// if we scrolled
```

Comment:
it can be one line:
      mAddCommentButton->setGeometry(commentBottonXpos, currSliderPos ? commentBottonYpos : commentBottonYpos + TOP_UNUSED_PIXELS_HEIGHT , side, side);

Reply:
fixed it